src/main/java/ViewModel/playlistViewModel.java

```diff
@@ -1,4 +1,37 @@
 package ViewModel;
+import Controller.newPlaylistController;
+import Entities.RegularUser;
+import Entities.Song;
+import InputData.newPlaylistInputData;
+import OutputData.newPlaylistOutputData;
+import UseCase.createPlaylist;
 
+
+import Presenter.playlistPresenter;
 public class playlistViewModel {
+    private final newPlaylistController newPlaylistController;
+    private  playlistPresenter playlistPresenter;
+
+    public playlistViewModel(int id, RegularUser user, String playlistName, Song firstSong){
+        // get necessary information from View to construct a new playlist
+        newPlaylistInputData inputData = new newPlaylistInputData(id,playlistName,firstSong,user);
+        //Blank output data object to extract success message from
+        newPlaylistOutputData outputData = new newPlaylistOutputData("");
+        this.playlistPresenter = new playlistPresenter(this, outputData);
+//        newPlaylistInputBoundary newPlaylistInputBoundary = InputBoundary.newPlaylistInputBoundary;
+        createPlaylist newPlay= new createPlaylist();
+        this.newPlaylistController = new newPlaylistController(newPlay,inputData);
+        //TODO: Does this violate clean architecture ?
```

Comment:
Not entirely sure if this is legal but I implemented the presenter/view stuff according to how Sarah described